pkg/util/validation/exporter/ring.go

```diff
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package exporter
+
+import (
+	"flag"
+	"fmt"
+	"hash/fnv"
+	"os"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/flagext"
+	"github.com/grafana/dskit/kv"
+	"github.com/grafana/dskit/netutil"
+	"github.com/grafana/dskit/ring"
+
+	util_log "github.com/grafana/mimir/pkg/util/log"
+)
+
+const (
+	// ringKey is the key under which we store the overrides-exporter's ring in the KVStore.
+	ringKey = "overrides-exporter"
+
+	// ringNumTokens is how many tokens each overrides-exporter should have in the ring.
+	// Overrides-exporters use a ring for discovery only, so one token is enough.
+	ringNumTokens = 1
```

Comment:
My assumption was that we don't care about balancing at all because in microservices mode we run on a single replica for overrides-exporter anyway(?). But you're right, sharding tenants in a somewhat balanced way can't hurt.